src/crap.nim

```diff
@@ -4,37 +4,35 @@ import
   strutils,
   times
 
-proc crap*(path: var string) =
-  var trashHome: string
+proc crap*(path: string) =
   let fullPath = expandFilename(path)
-  path = extractFilename(fullPath)
+  let fname = extractFilename(fullPath)
 
-  if existsEnv("XDG_DATA_HOME"):
-    trashHome = joinPath(getEnv("XDG_DATA_HOME"), "Trash")
+  let trashHome = if existsEnv("XDG_DATA_HOME"):
+    getEnv("XDG_DATA_HOME") / "Trash"
   else:
-    trashHome = joinPath(getHomeDir(), ".local/share/Trash")
+    getHomeDir() / ".local/share/Trash"
 
-  moveFile(fullPath, joinPath(trashHome, "files", path))
+  moveFile(fullPath, trashHome / "files" / fname)
 
   let 
     t = getTime()
-    formattedTime = t.format("yyyy-MM-dd") & "T" & t.format("HH:MM:ss")
+    formattedTime = t.format("yyyy-MM-dd'T'HH:MM:ss")
     trashInfo = fmt"""[Trash Info]
 Path={fullPath}
 DeletionDate={formattedTime}"""
 
-  writeFile(joinPath(trashHome, "info", fmt"{path}.trashinfo"), trashInfo)
+  writeFile(trashHome / "info" / fmt"{fname}.trashinfo", trashInfo)
 
 when isMainModule:
-  var del: string = ""
+  var del = ""
+  if paramCount() == 0:
+    echo("error: specify at least one file")
+    quit(1)
   try:
-    if paramCount() == 0:
-      echo("error: specify at least one file")
-      quit(0)
-    else:
-      for f in countUp(1, paramCount()):
-        del = paramStr(f).strip()
-        crap(del)
+    for f in countUp(1, paramCount()):
+      del = paramStr(f).strip()
+      crap(del)
   except OSError:
-    echo("error: no such file or directory")
-    quit(0)
+    echo("error: no such file or directory: " & del)
```

Comment:
Also added path to the file to the error which may help the user in case when moving fails